Project.toml

```diff
@@ -21,6 +20,5 @@ CommonSolve = "0.2"
 Distributions = "0.25"
 DistributionsAD = "0.6"
 SciMLBase = "1"
-StatsBase = "0.33"
```

Comment:
You can clean things up if you really want, but the whole dispatching/etc. will be rewritten to be as SciML compliant as possible once we have our code up and running.  So if this works, and passes tests, then I would consider just merging it , tagging, and worrying about cleanup later. 

Of course, if this package was making it difficult to tag/deploy to the other upstream pacakges then that is a different story and I understand completely.